plugins/modules/docker_secret.py

```diff
@@ -204,26 +226,51 @@ def __init__(self, client, results):
                 self.client.fail('Error while reading {src}: {error}'.format(src=data_src, error=to_native(exc)))
         self.labels = parameters.get('labels')
         self.force = parameters.get('force')
+        self.rolling_versions = parameters.get('rolling_versions')
+        self.versions_to_keep = parameters.get('versions_to_keep')
+
+        if self.rolling_versions:
+            self.version = 0
         self.data_key = None
+        self.secrets = []
 
     def __call__(self):
+        self.get_secret()
         if self.state == 'present':
             self.data_key = hashlib.sha224(self.data).hexdigest()
             self.present()
+            self.remove_old_versions()
         elif self.state == 'absent':
             self.absent()
 
+    def get_version(self, secret):
+        return int(secret.get('Spec', {}).get('Labels', {}).get('ansible_version', 0))
```

Comment:
```suggestion
        try:
            return int(secret.get('Spec', {}).get('Labels', {}).get('ansible_version', 0))
        except ValueError:
            return 0
```
In case someone adds their own `ansible_version` label with something that's not an integer :)

Reply:
Indeed, fixed in [c0ccb48](https://github.com/ansible-collections/community.docker/pull/293/commits/c0ccb48).